include/circt/Dialect/SMT/SMTOps.td

```diff
@@ -222,4 +222,64 @@ def DistinctOp : SMTOp<"distinct", [Pure, SameTypeOperands]> {
   let hasVerifier = true;
 }
 
+def IteOp : SMTOp<"ite", [
+  Pure,
+  AllTypesMatch<["thenValue", "elseValue", "result"]>
+]> {
+  let summary = "a if-then-else function";
+  let description = [{
+    This operation returns its second operand or its third operand depending on
+    whether its first operand is true or not. The semantics are equivalent to
+    the `ite` operator defined in the
+    [Core theory](https://smtlib.cs.uiowa.edu/Theories/Core.smt2) of the SMT-LIB
+    2.6 standard.
+  }];
+
+  let arguments = (ins BoolType:$cond,
+                       AnySMTType:$thenValue,
+                       AnySMTType:$elseValue);
+  let results = (outs AnySMTType:$result);
+
+  let assemblyFormat = [{
+    $cond `,` $thenValue `,` $elseValue attr-dict `:` qualified(type($result))
+  }];
+}
+
+class BoolOp<string mnemonic, string desc> : SMTOp<mnemonic, [Pure]> {
+  let summary = desc;
+  string explanation = [{
+    It supports a variadic number of operands, but requires at least two.
+    This is because the operator is annotated with the `:left-assoc` attribute
+    which means that `op a b c` is equivalent to `(op (op a b) c)`.
+  }];
+  let description = "This operation performs " # desc # [{.
+    The semantics are equivalent to the '}] # mnemonic # [{' operator in the
+    [Core theory](https://smtlib.cs.uiowa.edu/Theories/Core.smt2).
+    of the SMT-LIB Standard 2.6.
+    }] # explanation;
+
+  let arguments = (ins Variadic<BoolType>:$inputs);
+  let results = (outs BoolType:$result);
+
+  let assemblyFormat = "$inputs attr-dict";
+}
+
+def NotOp : BoolOp<"not", "a boolean negation"> {
+  let explanation = "";
+  let arguments = (ins BoolType:$input);
+  let assemblyFormat = "$input attr-dict";
+}
+
+def AndOp : BoolOp<"and", "a boolean conjunction">;
+def OrOp : BoolOp<"or", "a boolean disjunction">;
+def XOrOp : BoolOp<"xor", "a boolean exclusive OR">;
+
+def ImpliesOp : BoolOp<"implies", "boolean implication"> {
+  let explanation = [{
+    It supports a variadic number of operands, but requires at least two.
+    This is because the operator is annotated with the `:right-assoc` attribute
+    which means that `op a b c` is equivalent to `(op a (op b c))`.
+  }];
+}
```

Comment:
For this operation, I'm actually thinking if we should encode it as a variadic or not.
While for `and`, `or`, and `xor` it makes sense as the order does not matter, here the variadicity is just syntactic sugar for a right associative operation.
So to me, it would be better if this was only a binary operation, which is easier to reason about.